File: python/leetcode/etc/two_sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        first_idx = 0
        while first_idx < len(nums) - 1:
            first_element = nums[first_idx]

            for idx, num in enumerate(nums[first_idx + 1:]):
                if first_element + num == target:
                    return [first_idx, first_idx + idx + 1]

            first_idx += 1
        return None

File: go/leetcode/Hash Table/two_sum.go
// https://leetcode.com/problems/two-sum/submissions/

func twoSum(nums []int, target int) []int {
    result := []int{}
    lastIndex := len(nums)

    for i:=0; i<lastIndex; i++ {
        for j:= i+1; j<lastIndex; j++ {
            if nums[i] + nums[j] == target {
                result = append(result, i)
                result = append(result, j)
                return result
            }
        }
    }

    return result
}
